docs/configuration/hyperparameter_optimization.md

```diff
@@ -30,145 +33,86 @@ hyperopt:
+- `search_alg` section specifies the algorittm to sample the defined `parameters` space. Candidate algorithms are those found in [Ray Tune's Search Algorithms](https://docs.ray.io/en/latest/tune/api_docs/suggestion.html).
+- `executor` section specifies how to execute the hyperparameter optimization. The execution could happen locally in a serial manner or in parallel across multiple workers and with GPUs as well if available.  The `executor` section includes spefication for work scheduling and the number of samples to generate.
 
 # Defining hyperparameter search spaces
 
-In the `parameters` section, `.` is used to reference an parameter nested inside a section of the configuration.
-For instance, to reference the `learning_rate`, one would have to use the name `trainer.learning_rate`.
+In the `parameters` section, hyperparameters are dot( `.`) separate names. The parts of the hyperparameter names separated by `.` are references to nested sections in the Ludwig configuration.
+For instance, to reference the `learning_rate`, in the `trainer` section one would use the name `trainer.learning_rate`.
 If the parameter to reference is inside an input or output feature, the name of that feature will be be used as starting point.
 For instance, for referencing the `cell_type` of the `text` feature, use the name `text.cell_type`.
 
-## Number ranges
+## Numeric Hyperparameters
+
+- `space`: Use [Ray Tune's Search Space](https://docs.ray.io/en/latest/tune/api_docs/search_space.html) types, e.g., `uniform`, `quniform`, `loguniform`, `choice`, etc.  Refer the cited page for details.
 
-- `space`: Use `linear` or `log`.
-- `range.low`: the minimum value the parameter can have
-- `range.high`: the maximum value the parameter can have
-- `steps`: (optional) number of steps to break down a range.
+For numeric `spaces`, these define the range where the value is generated
 
-For instance `range: (0.0, 1.0), steps: 3` would yield `[0.0, 0.5, 1.0]` as potential values to sample from, while if
-`steps` is not specified, the full range between `0.0` and `1.0` will be used.
+- `lower`: the minimum value the parameter can have
+- `upper`: the maximum value the parameter can have
+- `q`: quantization number, used in `spaces` such as `quniform`, `qloguniform`, `qrandn`, `qrandint`, `qlograndint`
+- `base`: defines the base of the log for `loguniform`, `qloguniform`, `lograndint` and `qlograndint`
 
-Float example:
+**Float example**: Uniform floating point random values (in log space) between 0.001 and 0.1
 
 ```yaml
 trainer.learning_rate:
-  space: linear
-  range:
-    low: 0.001
-    high: 0.1
-  steps: 4
+  space: loguniform
+  lower: 0.001
+  uppper: 0.1
 ```
 
-Integer example:
+**Integer example**: Uniform random integer values 1, 2, 3
```

Comment:
It seems that for different spaces upper can be inclusive or exclusive. As it is weird that the behavior changes maybe adding a note (that renders as a note) suggesting checking in the ray documentation if ranges are inclusive or exclusive would be valuable

Reply:
done